storage/StorageBlock.cpp

```diff
@@ -1361,4 +1361,9 @@ void StorageBlock::invalidateAllIndexes() {
   }
 }
 
+const std::size_t StorageBlock::getNumTuples() const {
```

Comment:
If we adopt my suggestions, we don't need this new method. We would have access to `tuple_store`.